src/walk.rs

```diff
@@ -27,9 +28,13 @@ fn walk(tx: channel::Sender<Message>, entries: &[PathBuf], filesize_type: Filesi
 
             let size = filesize_type.size(&metadata);
 
-            tx_ref.send(Message::SizeEntry(unique_id, size)).unwrap();
+            if metadata.is_file() {
+                tx_ref.send(Message::SizeEntry(unique_id, size)).unwrap();
+            }
 
             if metadata.is_dir() {
+                tx_ref.send(Message::DirectoryEntry(size)).unwrap();
```

Comment:
so we don't send `unique_id`s for directories? does that really work? we should probably have added tests for this.

Reply:
Very possible I don't fully grasp what they we're supposed to do, but my understanding was that unique ID's were used to prevent double-counting of files that had hard links.  From what I've read, hard links aren't possible for directories so the case shouldn't exist where we need to worry about counting them multiple times.

Good chance I'm wrong though, so let me know if there's a reason to add that.  I couldn't really test the case on my local system since I can't create directory hard links in Ubuntu at least.

Reply:
You are right, there don't seem to be hardlinks for directories.

But I think we should add a few more tests to make sure that we get this right. There are probably a lot of edge cases (with symbolic links as well). I know that the test suite is not very complete so far, but this could be a good start :smile:.

Let me know if this drags on too long for you and we can probably find someone else to finish this.